libs/generic-view/store/src/lib/entities/reducer.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { createReducer } from "@reduxjs/toolkit"
+import { EntitiesConfig, EntitiesMetadata, EntityData } from "device/models"
+import {
+  setEntitiesConfig,
+  setEntitiesData,
+  setEntitiesMetadata,
+  setEntityData,
+} from "./actions"
+
+type EntityType = string
+
+interface Entities {
+  idFieldKey?: string
+  config?: EntitiesConfig
+  data?: EntityData[]
+  metadata?: EntitiesMetadata
+}
+
+interface EntitiesState {
+  entities: {
+    [key: EntityType]: Entities | undefined
+  }
+}
+
+const initialState: EntitiesState = {
+  entities: {},
+}
+
+export const genericEntitiesReducer = createReducer(initialState, (builder) => {
+  builder.addCase(setEntitiesConfig, (state, action) => {
+    state.entities[action.payload.entityType] = {
+      ...state.entities[action.payload.entityType],
+      config: action.payload.config,
+      idFieldKey: action.payload.idFieldKey,
+    }
+  })
+  builder.addCase(setEntitiesData, (state, action) => {
+    state.entities[action.payload.entityType] = {
+      ...state.entities[action.payload.entityType],
+      data: action.payload.data,
+    }
+  })
+  builder.addCase(setEntityData, (state, action) => {
+    state.entities[action.payload.entityType]?.data?.map((entity) => {
+      if (
+        entity[state.entities.idFieldKey as keyof typeof entity] ===
```

Comment:
Shouldn't idFieldKey be accessed via state.entities[action.payload.entityType]?.idFieldKey?

```suggestion
        state.entities[action.payload.entityType]?.idFieldKey] ===
```

Reply:
It should 😉 Fixed in [1c1a314](https://github.com/mudita/mudita-center/pull/2049/commits/1c1a314cf61b453ead2db9a88e1e811af84cb23c)